### ISSUE TITLE
tsql: GRANT CONTROL TO [something] unparsable

### ISSUE THREAD

**Issue by ddbtrmatic** [sqlfluff/sqlfluff#7848]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This syntax is unparsable using the tsql dialect

```sql
GRANT CONTROL TO [something]
GO
```

### Expected Behaviour

`GRANT CONTROL TO [something]` is valid syntax. This query grants the CONTROL permission to the database user or role.

### Observed Behaviour

```
INFO
###
#
# Fixed Tree:
#
###
INFO
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'BatchSegment'>]>]>"
[L:  1, P:  1]      |        word:                                                 'GRANT'
[L:  1, P:  6]      |        whitespace:                                           ' '
[L:  1, P:  7]      |        word:                                                 'CONTROL'
[L:  1, P: 14]      |        whitespace:                                           ' '
[L:  1, P: 15]      |        word:                                                 'TO'
[L:  1, P: 17]      |        whitespace:                                           ' '
[L:  1, P: 18]      |        square_quote:                                         '[something]'
[L:  1, P: 29]      |        newline:                                              '\n'
[L:  2, P:  1]      |        word:                                                 'GO'
[L:  2, P:  3]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    [META] end_of_file:

== [grant.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'GRANT
                       | CONTROL TO [something]\nGO'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

### How to reproduce

`echo "GRANT CONTROL TO [something]" > grant.sql && sqlfluff lint --dialect tsql grant.sql`

### Dialect

tsql

### Version

❯ sqlfluff --version
sqlfluff, version 4.2.0

### Configuration

No config provided, default.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @ddbtrmatic with the latest `sqlfluff` this parses properly:

```bash
sqlfluff parse test.sql -d tsql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    batch:
[L:  1, P:  1]      |        statement:
[L:  1, P:  1]      |            grant_statement:
[L:  1, P:  1]      |                keyword:                                      'GRANT'
[L:  1, P:  6]      |                whitespace:                                   ' '
[L:  1, P:  7]      |                permissions_segment:
[L:  1, P:  7]      |                    keyword:                                  'CONTROL'
[L:  1, P: 14]      |                whitespace:                                   ' '
[L:  1, P: 15]      |                keyword:                                      'TO'
[L:  1, P: 17]      |                whitespace:                                   ' '
[L:  1, P: 18]      |                role_reference:
[L:  1, P: 18]      |                    quoted_identifier:                        '[something]'
[L:  1, P: 29]      |    [META] end_of_file:
```

**Comment by ddbtrmatic** [comment]
@peterbud I can confirm this is fixed. Thank you!
